src/tree/EnvironmentTreeItem.ts

```diff
@@ -96,4 +100,10 @@ export class EnvironmentTreeItem extends AzureParentTreeItem implements IAzureRe
 
         return undefined;
     }
+
+    public async refreshImpl(): Promise<void> {
+        const remote: string | undefined = await tryGetRemote();
+        const branch: string | undefined = remote ? await tryGetBranch() : undefined;
+        this.inWorkspace = this.parent.data.properties.repositoryUrl === remote && this.data.properties.sourceBranch === branch;
```

Comment:
You're doing this here and in `loadMoreChildrenImpl`. Can we share that code somehow?

```
this.parent.data.properties.repositoryUrl === remote && this.data.properties.sourceBranch === branch;
```

Reply:
I don't know if we even need this tbh.  I did it in case we ended up supporting multi-root, but you wouldn't be able to open a new folder without reactivating the extension so it'd have to regenerate the entire tree anyway.

Reply:
Never mind, I remember why I thought this was necessary now.  Users can easily change the branch of their project so we at least need a refresh so we can reflect that.